### PR TITLE
feat: support for creating offline tables

### PR DESCRIPTION
**Pull request by laxmanchekka**
## Description
Create OFFLINE tables as part of the view creation

### Testing
- Added unit tests
- Tested on local single node cluster

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotTableSpec.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  private String timeColumn;
+  private String retentionTimeValue;
+  private String retentionTimeUnit;
+  private String segmentAssignmentStrategy;
+
+  // Tenants config
+  private String brokerTenant;
+  private String serverTenant;
+
+  public PinotTableSpec() {}
```
Comment (surajpuvvada) [inline view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotTableSpec.java:47]: u could use a @NoArgsConstructor lombok annotation

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotViewCreatorConfig.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package org.hypertrace.core.viewcreator.pinot;
+
+public class PinotViewCreatorConfig {
```
Comment (surajpuvvada) [inline view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotViewCreatorConfig.java:3]:
```suggestion
public class PinotViewCreatorConfigConstants {
```

File: view-creator-framework/src/test/resources/sample-view-generation-spec.conf [hunk 3]
```diff
@@ -1,18 +1,12 @@
...
 }
 
-kafka = {
-  brokerAddress = "localhost:9092"
-  topicName = test-view-events
-  partitions = 8
-  replicationFactor = 3
-}
-
 pinot = {
```
Comment (surajpuvvada) [inline view-creator-framework/src/test/resources/sample-view-generation-spec.conf:10]: we don't need a separate block for offline config ?

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotTableSpec.java (hunk 1 above)
Comment (laxmanchekka) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotTableSpec.java]: Fixed.

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotViewCreatorConfig.java (hunk 2 above)
Comment (laxmanchekka) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotViewCreatorConfig.java:3]: We are using Config as Suffix for such constants (JobConfig)

File: view-creator-framework/src/test/resources/sample-view-generation-spec.conf (hunk 3 above)
Comment (laxmanchekka) [reply view-creator-framework/src/test/resources/sample-view-generation-spec.conf:10]: its optional. added it anyway and added tests for it.

**Review by kotharironak — APPROVED** [review]

File: view-creator-framework/src/test/resources/sample-view-generation-spec.conf [hunk 4]
```diff
@@ -1,18 +1,12 @@
 tool.class = org.hypertrace.core.viewcreator.pinot.PinotTableCreationTool
 service.name = "sample-view"
+
 view = {
   name = myView
   output.schema.class = org.hypertrace.core.viewcreator.test.api.TestView
   output.schema.url = "host:port/myView"
 }
 
-kafka = {
```
[the remark this answers is not in the document]
Comment (laxmanchekka) [reply view-creator-framework/src/test/resources/sample-view-generation-spec.conf:9]: No. Its unused. Not required any more.

**Review by ravisingal — APPROVED** [review]

**Pull request merged**